packages/tooling/fast-tooling/src/data-utilities/message-system-utility-action.ts

```diff
@@ -0,0 +1,50 @@
+export interface MessageSystemUtilityActionCallbackConfig {
+    /**
+     * The unique identifier
+     */
+    id: string;
+}
+
+export interface MessageSystemUtilityActionConfig
+    extends MessageSystemUtilityActionCallbackConfig {
+    /**
+     * The action to take when the keycodes have been pressed
+     */
+    action: () => void;
+}
+
+/**
+ * This abstract class is for actions intended to be part
+ * of a registered class
+ *
+ * It takes two generics, C, an object which is added to the
+ * MessageSystemUtilityActionCallbackConfig,
+ * and M, a generic used for matching
+ * the action to some specific parameters
+ */
+export abstract class MessageSystemUtilityAction<C = {}, M = {}> {
```

Comment:
I haven't been totally consistent about this in the past, but looking at this I'm reminded of how I've sometimes named generic parameters to make them clearer. I'll sometimes use a T prefix and then provide a meaningful name. So in this case, you might have `TConfig` and `TMatchable` or something like that.